pkg/logs/input/journald/tailer.go

```diff
@@ -0,0 +1,157 @@
+package journald
+
+import (
+	"encoding/json"
+	"io"
+	"strings"
+	"time"
+
+	log "github.com/cihub/seelog"
+	"github.com/coreos/go-systemd/sdjournal"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+)
+
+// defaultWaitDuration represents the delay before which we try to collect a new log from the journal
+const defaultWaitDuration = 1 * time.Second
+
+// JournalConfig enables to configure the tailer:
+// - Units: the units to filter on
+// - Path: the path of the journal
+type JournalConfig struct {
+	Units []string
+	Path  string
+}
+
+// Tailer collects logs from a journal.
+type Tailer struct {
+	config     JournalConfig
+	source     *config.LogSource
+	outputChan chan message.Message
+	journal    *sdjournal.Journal
+	stop       chan struct{}
+}
+
+// NewTailer returns a new tailer.
+func NewTailer(config JournalConfig, source *config.LogSource, outputChan chan message.Message) *Tailer {
+	return &Tailer{
+		config:     config,
+		source:     source,
+		outputChan: outputChan,
+		stop:       make(chan struct{}, 1),
+	}
+}
+
+// setup configures the tailer
+func (t *Tailer) setup() error {
+	config := t.config
+	var err error
+
+	if config.Path == "" {
+		// open the default journal
+		t.journal, err = sdjournal.NewJournal()
+	} else {
+		t.journal, err = sdjournal.NewJournalFromDir(config.Path)
+	}
+	if err != nil {
+		return err
+	}
+
+	for _, unit := range config.Units {
```

Comment:
Question: do we allow to filter only on `unit`, or do we let the user set himself the match value?
ie in the future would we want to filter only on unit or on more?